File: Daily Challanges/2026-04-April/2026-04-29.js
function parseUrlQuery(url) {
  const queryString = url.split("?")[1];
  const pairs = queryString.split("&");
  let dictionary = {};
  for (const pair of pairs) {
    const [key, value] = pair.split("=");
    dictionary[key] = value;
  }
  return dictionary;
}

console.log(parseUrlQuery("https://example.com/search?name=Alice&age=30"));
console.log(
  parseUrlQuery(
    "https://freecodecamp.org/learn?skill=programming&language=python",
  ),
);

File: Daily Challanges/2026-04-April/2026-04-29.py
def parse_url_query(url):
    query_string = url.split("?")[1]
    pairs = query_string.split("&")
    dictionary = {}
    for pair in pairs:
        key, value = pair.split("=")
        dictionary[key] = value
    return dictionary


print(parse_url_query("https://example.com/search?name=Alice&age=30"))
print(
    parse_url_query("https://freecodecamp.org/learn?skill=programming&language=python")
)
print(parse_url_query("https://freecodecamp.org/items?category=books&sort=asc&page=2"))
